### PR TITLE
Built functionality for creating and reading venues, created venues db table

### PR DESCRIPTION
**Pull request by rwhitten577**

### CODE REVIEW

File: app/controllers/venues_controller.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  def index
+    @venues = Venue.all.order("name ASC")
+
+    if @venues.empty?
+      flash[:notice] = "There are no venues"
+    end
+  end
+
+  def show
+    @venue = set_venue
```
Comment (justinhuynh) [inline app/controllers/venues_controller.rb]: FYI this is really only useful if you're calling `set_venue` multiple times. Otherwise, you can just do `@venue = Venue.find(params[:id])`

File: app/helpers/states_helper.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
+module StatesHelper
+
+def us_states
```
Comment (justinhuynh) [inline app/helpers/states_helper.rb]: Make this a constant instead of a method. so `US_STATES =  [`

File: app/helpers/states_helper.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    ['Texas', 'TX'],
+    ['Utah', 'UT'],
+    ['Vermont', 'VT'],
+    ['Virginia', 'VA'],
+    ['Washington', 'WA'],
+    ['West Virginia', 'WV'],
+    ['Wisconsin', 'WI'],
+    ['Wyoming', 'WY']
+  ]
+end
```
Comment (justinhuynh) [inline app/helpers/states_helper.rb]: indentation needs to be aligned

File: app/models/venue.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+class Venue < ActiveRecord::Base
+  validates :name, presence: true
+  validates :street, presence: true
+  validates :city, presence: true
+  validates :state, presence: true, length: { is: 2 }
+  validates :category, presence: true, inclusion: { in: %w(Music Comedy Poetry Theater) }
```
Comment (justinhuynh) [inline app/models/venue.rb]: nice!

File: app/views/layouts/application.html.erb [hunk 5]
```diff
@@ -8,6 +8,14 @@
 </head>
 <body>
 
+  <% if !flash[:notice].nil? %>
```
Comment (justinhuynh) [inline app/views/layouts/application.html.erb]: Presumably you will want to show `flash[:errors]` as well. This would not get the job done, since it only shows `flash[:notice]`. Instead, try something like this:

```
  <div class="container">
   <% flash.each do |key, value| %>
     <div class="alert alert-<%= key %>"><%= value %></div>
   <% end %>
  </div>
```

File: app/controllers/venues_controller.rb [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  end
+
+  def create
+    @venue = Venue.new(venue_params)
+
+    if @venue.save
+      flash[:notice] = "Venue successfully added!"
+      redirect_to @venue
+    else
+      flash[:notice] = "There were problems saving your venue."
```
Comment (justinhuynh) [inline app/controllers/venues_controller.rb]: If you update the `flash` snippet per my comment below, you will be able to use other `flash` messages. 

The logic that is currently in your view should actually be in here:

`flash[:errors] = @venue.errors.full_messages.join(", ")`

File: spec/features/creates_venue_spec.rb [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+
+
+    expect(page).to have_content("Name can't be blank")
+    expect(page).not_to have_content("Venue successfully added!")
+    expect(page).not_to have_content("Street can't be blank")
+    expect(page).not_to have_content("City can't be blank")
+    expect(page).not_to have_content("State can't be blank")
+    expect(page).not_to have_content("Category can't be blank")
+    expect(page).to have_content("There were problems saving your venue")
+    expect(find("#venue_street").value).to eq(venue[:street])
```
Comment (justinhuynh) [inline spec/features/creates_venue_spec.rb]: put a space between blocks of test code to help break things up (between line 67 and 68 is a good spot)

File: spec/features/creates_venue_spec.rb [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+    expect(page).not_to have_content("Category can't be blank")
+    expect(page).to have_content("There were problems saving your venue")
+    expect(find("#venue_name").value).to eq(venue[:name])
+    expect(find("#venue_city").value).to eq(venue[:city])
+    expect(find("#venue_website").value).to eq(venue[:website])
+    expect(find("#venue_category_music").value).to eq(venue[:category])
+    expect(find("#venue_description").value).to eq(venue[:description])
+  end
+
+  scenario "user does not input city" do
```
Comment (justinhuynh) [inline spec/features/creates_venue_spec.rb]: This is nice and through, and it's also well organized. However, we don't really need to test every last field/error/validation. Rather, we want to test that the user maybe leaves some fields blank and sees pertinent errors. Testing the edge cases of whether valid input is accepted can stay in your model tests

File: spec/features/creates_venue_spec.rb [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+    choose("venue_category_music")
+    fill_in("venue_description", with: venue[:description])
+    click_button("Add Venue")
+
+    expect(current_path).to eq("/venues/#{Venue.first.id}")
+    expect(page).to have_content("Venue successfully added!")
+    expect(page).to have_content(venue[:name])
+  end
+
+  scenario "user does not input name" do
```
Comment (justinhuynh) [inline spec/features/creates_venue_spec.rb]: I would ditch the individual error cases and generalize here. Again - this level of detail will be covered by your model tests (which test the model validations that give you the error messages in the first place) So something like this, where you test a couple fields being blank:

```
scenario "user does not complete required fields" do
    visit venues_path
    click_link "Add New Venue" 
    fill_in "Name", with: "" # here you're being explicit that the field Name is being left blank
    fill_in "Street", with: "" # here you're being explicit that the field Name is being left blank
    fill_in "City", with: venue[:city]
    select "Massachusetts", from: "State"
    fill_in "Venue Website", with: venue[:website]
    choose "venue_category_music" 
    fill_in "venue_description", with: venue[:description]
    click_button "Add Venue"


    expect(page).to have_content("Name can't be blank")
    expect(page).to have_content("Street can't be blank")

    expect(page).not_to have_content("Venue successfully added!")

    expect(page).to have_content("There were problems saving your venue")

    expect(find("#venue_city").value).to eq(venue[:city])
    expect(find("#venue_website").value).to eq(venue[:website])
    expect(find("#venue_category_music").value).to eq(venue[:category])
    expect(find("#venue_description").value).to eq(venue[:description])
```

File: spec/features/sees_venues_spec.rb [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+# Acceptance Criteria
+#
+# - I must see the name of each venue
+# - I must see venues listed in alphabetical order
+
+
+feature "user sees venues" do
+  scenario "see all the venues" do
+    venue = FactoryGirl.create(:venue, name: 'ZZZZ')
+    another_venue = FactoryGirl.create(:venue, name: 'AAAA')
```
Comment (justinhuynh) [inline spec/features/sees_venues_spec.rb]: minor thing - be consistent about using single or double quotes

File: spec/rails_helper.rb [hunk 11]
```diff
@@ -54,4 +55,7 @@
   config.filter_rails_from_backtrace!
   # arbitrary gems may also be filtered via:
   # config.filter_gems_from_backtrace("gem name")
```
Comment (justinhuynh) [inline spec/rails_helper.rb]: ditch the auto-generated comments

File: spec/spec_helper.rb [hunk 12]
```diff
@@ -1,5 +1,6 @@
 require 'coveralls'
 Coveralls.wear!('rails')
+require 'support/factory_girl'
 
 # This file was generated by the `rails generate rspec:install` command. Conventionally, all
 # specs live under a `spec` directory, which RSpec adds to the `$LOAD_PATH`.
```
Comment (justinhuynh) [inline spec/spec_helper.rb]: ditch the auto-generated comments
